submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/ExtSubmissionSubmitter.kt

```diff
@@ -0,0 +1,57 @@
+package ac.uk.ebi.biostd.submission.submitter
+
+import ac.uk.ebi.biostd.persistence.common.request.SubmissionRequest
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionDraftService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceService
+import ac.uk.ebi.biostd.persistence.filesystem.service.FileProcessingService
+import ac.uk.ebi.biostd.submission.model.ReleaseRequest
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.FireFile
+import ebi.ac.uk.extended.model.NfsFile
+import ebi.ac.uk.io.ext.md5
+import ebi.ac.uk.io.ext.size
+
+class ExtSubmissionSubmitter(
+    private val persistenceService: SubmissionPersistenceService,
+    private val draftService: SubmissionDraftService,
+    private val requestProcessor: RequestProcessor,
+) {
+    fun submitAsync(request: SubmissionRequest): Pair<String, Int> {
+        return saveRequest(request, request.submission.submitter)
+    }
+
+    fun processRequest(accNo: String, version: Int): ExtSubmission {
+        val sub = requestProcessor.loadRequest(accNo, version)
+        return persistenceService.processSubmissionRequest(sub)
+    }
+
+    fun release(request: ReleaseRequest) {
+        val (accNo, owner, relPath) = request
+        persistenceService.releaseSubmission(accNo, owner, relPath)
+    }
+
+    private fun saveRequest(request: SubmissionRequest, owner: String): Pair<String, Int> {
+        val saved = persistenceService.saveSubmissionRequest(request)
+        request.draftKey?.let { draftService.setProcessingStatus(owner, it) }
+        return saved
+    }
+}
+
+class RequestProcessor(
```

Comment:
I think this should be on its own file since it's a functionality class. I understand these two have kind of the same business but I feel it would make more sense.

Also, I would suggest naming it `SubmissionRequestProcessor` since we have several kind of requests on the code and finding the name `RequestProcessor`, by itself, doesn't really give me a clue of what it does